test/test_data_lake.py

```diff
@@ -19,26 +19,91 @@
 
 sys.path[0:0] = [""]
 
-from test import client_context, unittest
+from pymongo.auth import MECHANISMS
+from test import client_context, unittest, IntegrationTest
 from test.crud_v2_format import TestCrudV2
-from test.utils import TestCreator
+from test.utils import (
+    rs_or_single_client, OvertCommandListener, TestCreator,
+    _mongo_client)
 
 
 # Location of JSON test specifications.
 _TEST_PATH = os.path.join(
     os.path.dirname(os.path.realpath(__file__)), "data_lake")
 
 
+class TestDataLakeMustConnect(IntegrationTest):
+    def test_connected_to_data_lake(self):
+        data_lake = os.environ.get('DATA_LAKE')
+        if not data_lake:
+            self.skipTest('DATA_LAKE is not set')
+
+        self.assertTrue(client_context.is_data_lake,
+                        'client context.is_data_lake must be True when '
+                        'DATA_LAKE is set')
+
+
+class TestDataLakeProse(IntegrationTest):
+    # Default test database and collection names.
+    TEST_DB = 'test'
+    TEST_COLLECTION = 'driverdata'
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestDataLakeProse, cls).setUpClass()
+        if not client_context.is_data_lake:
+            raise unittest.SkipTest('Not connected to Atlas Data Lake')
+
+    # Test killCursors
+    def test_1(self):
+        listener = OvertCommandListener()
+        client = rs_or_single_client(event_listeners=[listener])
+        cursor = client[self.TEST_DB][self.TEST_COLLECTION].find(
+            {}, batch_size=2)
+        next(cursor)
+
+        # find command assertions
+        find_cmd = listener.results["succeeded"][-1]
+        self.assertEqual(find_cmd.command_name, "find")
+        cursor_id = find_cmd.reply["cursor"]["id"]
+        cursor_ns = find_cmd.reply["cursor"]["ns"]
+
+        # killCursors command assertions
+        cursor.close()
+        started = listener.results["started"][-1]
+        self.assertEqual(started.command_name, 'killCursors')
+        succeeded = listener.results["succeeded"][-1]
+        self.assertEqual(succeeded.command_name, 'killCursors')
+
+        self.assertIn(cursor_id, started.command["cursors"])
+        target_ns = ".".join([started.command['$db'],
+                              started.command['killCursors']])
+        self.assertEqual(cursor_ns, target_ns)
+
+        self.assertIn(cursor_id, succeeded.reply["cursorsKilled"])
+
+    # Test no auth
+    def test_2(self):
+        client = _mongo_client(None, None, authenticate=False)
+        client.admin.command('ping')
+
+    # Test with auth
+    def test_3(self):
+        for mechanism in ['SCRAM-SHA-1', 'SCRAM-SHA-256']:
+            client = rs_or_single_client(authMechanism=mechanism)
+            client.admin.command('ping')
+
+
 class DataLakeTestSpec(TestCrudV2):
     # Default test database and collection names.
     TEST_DB = 'test'
     TEST_COLLECTION = 'driverdata'
 
     @classmethod
-    @unittest.skipUnless(client_context.is_data_lake,
```

Comment:
Please use `client_context._require` as a decorator instead. Consider renaming to `client_context.require`. 

The problem with the new code is that it leaves client_knobs enabled because when setUpClass errors with an exception then tearDownClass is never called.

Reply:
Ah, right... Added a new helper `client_context.require_data_lake`